src/lib/transition_frontier/README.md

```diff
@@ -0,0 +1,156 @@
+# Transition Frontier
+
+The transition frontier is hybrid in-memory/on-disk data structure which represents all known states on the network up to the point of finality. This data structure plays an essential role in tracking states produced by consensus protocol in a way that automatically garbage collects orphaned states which will never become part of the canonical chain, and does so in a way that optimizes for high read/write performance while also persisting information in the background for future recovery. As such, the transition frontier can be thought of as both an in-memory data structure, and a concurrent subsystem which maintains a persistent on-disk copy of the datastructure. To helpp maintain this viewpoint, the implementation is split into 2 main parts: a full frontier, which stores the entire expanded state of each block in-memory, and a persistent frontier, which asynchronously processes state transitions to the full frontier, proxying those operations to a RocksDB representation of the frontier in the background.
```

Comment:
Should we split the first sentence to its own line, so it's really clear whether this is 'the library you're looking for'? at a glance So like

```
The transition frontier represents all known states on the network up to the point of finality.

This hybrid in-memory/on-disk data structure ...
```